java/vector/src/main/java/org/apache/arrow/vector/complex/impl/PromotableWriter.java

```diff
@@ -301,13 +302,15 @@ public boolean isEmptyStruct() {
     return writer.isEmptyStruct();
   }
 
+  @Override
   protected FieldWriter getWriter() {
     return writer;
   }
 
   private FieldWriter promoteToUnion() {
     String name = vector.getField().getName();
-    TransferPair tp = vector.getTransferPair(vector.getMinorType().name().toLowerCase(), vector.getAllocator());
+    TransferPair tp = vector.getTransferPair(vector.getMinorType().name().toLowerCase(Locale.getDefault()),
+        vector.getAllocator());
```

Comment:
Hmm, setting aside the question of why we lowercase in the first place, we should use Locale.ROOT to be consistent?

Reply:
Right! I had doubts between the right one `ROOT` or `getDefault()`, let's use `ROOT`